api/src/main/java/org/apache/iceberg/Scan.java

```diff
@@ -171,4 +172,10 @@ default ThisT select(String... columns) {
 
   /** Returns the split open file cost for this scan. */
   long splitOpenFileCost();
+
+  /** Create a new scan that will report the scan metrics to the {@code reporter} */
+  default ThisT metricsReporter(MetricsReporter reporter) {
+    throw new UnsupportedOperationException(
+        this.getClass().getName() + " doesn't implement metricReporter");
```

Comment:
Typo in the method name inside the comment (should be `metricsReporter`).

Reply:
Looks like this comment was missed. The exception message should include the correct method name.